tracecat/expressions/expectations.py

```diff
@@ -87,10 +96,42 @@ def reference_type(self, name) -> str:
         logger.trace("Reference type:", name=name)
         return f"${name.value}"
 
+    @v_args(inline=True)
+    def enum_type(self, *values) -> Enum:
+        if len(values) > self.MAX_ENUM_VALUES:
+            raise ValueError(f"Too many enum values (maximum {self.MAX_ENUM_VALUES})")
+
+        enum_values = {}
+        seen_values = set()
+
+        for value in values:
+            if not value:
+                raise ValueError("Enum value cannot be empty")
+
+            # Case-insensitive duplicate check
+            value_lower = value.lower()
+            if value_lower in seen_values:
+                raise ValueError(f"Duplicate enum value: {value}")
+
+            seen_values.add(value_lower)
+            enum_values[value] = value
+
+        # Convert to upper camel case (e.g., "user_status" -> "UserStatus")
+        enum_name = "".join(word.title() for word in self.field_name.split("_"))
+        logger.trace("Enum type:", name=enum_name, values=enum_values)
+        return Enum(enum_name, enum_values)
 
-def parse_type(type_string: str) -> Any:
+    @v_args(inline=True)
+    def STRING_LITERAL(self, value) -> str:
+        # Remove quotes from the value
+        value = value.strip('"').strip("'")
+        # Coerce to string
+        return str(value)
```

Comment:
is this not already a string if you can strip it?

Reply:
Oh yeah.